src/sw.js

```diff
@@ -0,0 +1,139 @@
+    let cacheResponse = await caches.match(event.request);
+    let request = event.request;
+    if (cacheResponse !== undefined) {
+      // cache hit
+      return cacheResponse;
+    }
+    // miss
+    else {
+      return fetch(request).then(response => {
+        // response can only be consumed once
+        // so save clone to put copy in cache
+        let responseClone = response.clone();
+        caches.open(caching.cacheName).then(cache =>
+          cache.put(request, responseClone)
+        );
+        return response;
+      });
+    }
+  },
+
+  networkFirst: async function (event) {
+    let request = event.request;
+    try {
+      return fetch(request).then(
+        response => {
+        // update cache after network fetch
+          let responseClone = response.clone();
+          caches.open(caching.cacheName).then(cache =>
+            cache.put(request, responseClone)
+          );
+          return response;
+        });
+    }
+    catch(err) {
+      return caches.match(event.request);
+    }
+  },
+
+  networkOnly: async function(event) {
+    return fetch(event.request);
+  }, 
+  
+  raceCacheNetwork: async function(event) {
+    let request = event.request;
+    
+    let cachePromise = caches.match(request).then(
+      response => {
+        if (response) {
+          return response;
+        }
+        else
+          throw Error('cache miss');
+      }
+    );
+    
+    let networkPromise = fetch(request).then(
+      response => {
+        let responseClone = response.clone();
+        caches.open(caching.cacheName).then(cache =>
+          cache.put(request, responseClone)
+        );
+        return response;
+      }
+    );
+
+    return Promise.any([cachePromise, networkPromise]);
+  }
+};
+
+meta.policies = [
+  {
+    // images
+    url: /^(ftp|https?):.*\.(jpe?g|png|gif|svg)($|\?.*)/i,
+    handle: caching.strategies.cacheFirst
+  },
+  {
+    // audios & videos
+    url: /^(ftp|https?):.*\.(mp\d|webm|ogg|wav|flac)($|\?.*)/i,
+    handle: caching.strategies.cacheFirst
+  },
+  {
+    // fonts
+    url: /^(ftp|https?):.*\.(ttf|otf|woff\d?)($|\?.*)/i,
+    handle: caching.strategies.cacheFirst
+  },
+  {
+    // web content
+    url: /^(ftp|https?):.*\.([jt]sp?|css|html?)($|\?.*)/i,
+    handle: caching.strategies.networkFirst
+  },
+  {
+    // web data
+    url: /^(ftp|https?):.*\.(csv|json|txt|xml)($|\?.*)/i,
+    handle: caching.strategies.raceCacheNetwork
+  },
+  // fallback
+  {
+    url: /.+/i,
+    handle: caching.strategies.networkOnly
+  }
+];
+
+self.addEventListener('install', event => {
+  let cacheAllowlist = [caching.cacheName];
+
+  console.log('[ServiceWorker] installed');
+  event.waitUntil(
+    caches.keys().then(function (keyList) {
+      return Promise.all(keyList.map(function (key) {
+        if (cacheAllowlist.indexOf(key) === -1) {
+          console.log('[ServiceWorker] Removing old cache ', key);
+          return caches.delete(key);
+        }
+      }));
+    })
+  );
```

Comment:
```suggestion
  event.waitUntil(async () => {
    const keyList = await caches.keys();
    const promises = keyList.map(key => {
      if (cacheAllowlist.indexOf(key) === -1) {
        console.log('[ServiceWorker] Removing old cache ', key);
        return caches.delete(key);
      }
    })
    return Promise.all(promises)
  });
```
Just a readability improvement.

Reply:
Got it. Thanks.